build_bins.sh

```diff
@@ -1,354 +1,90 @@
-  gem install json --no-ri --no-rdoc -v 1.8.3
-fi
-
-if [[ -z `gem list --local cabin | grep cabin | cut -f1 -d" "` ]]; then
-  gem install cabin --no-ri --no-rdoc -v 0.7.2
-fi
-
-if [[ -z `gem list --local fpm | grep fpm | cut -f1 -d" "` ]]; then
-  gem install fpm --no-ri --no-rdoc -v 1.3.3
-fi
-
-# Download jmxtrans tar.gz file
-if ! [[ -f jmxtrans-256-dist.tar.gz ]]; then
-  while ! $(file jmxtrans-256-dist.tar.gz | grep -q 'gzip compressed data'); do
-    $CURL -O -L -k http://central.maven.org/maven2/org/jmxtrans/jmxtrans/256/jmxtrans-256-dist.tar.gz
-   done
-fi
-FILES="jmxtrans-254-dist.tar.gz $FILES"
-
-# Fetch Kafka Tar
-for version in 0.9.0.1; do
-  mkdir -p kafka/${version}/
-  if ! [[ -f kafka/${version}/kafka_2.11-${version}.tgz ]]; then
-    pushd kafka/${version}/
-    while ! $(file kafka_2.11-${version}.tgz | grep -q 'gzip compressed data'); do
-      $CURL -O -L http://mirrors.ocf.berkeley.edu/apache/kafka/${version}/kafka_2.11-${version}.tgz
-    done
-    popd
-  fi
-  FILES="kafka_2.11-${version}.tgz $FILES"
-done
-
-# Fetch Java Tar
-if ! [[ -f jdk-8u101-linux-x64.tar.gz ]]; then
-  while ! $(file jdk-8u101-linux-x64.tar.gz | grep -q 'gzip compressed data'); do
-    $CURL -O -L -C - -b "oraclelicense=accept-securebackup-cookie" http://download.oracle.com/otn-pub/java/jdk/8u101-b13/jdk-8u101-linux-x64.tar.gz
-  done
-fi
-FILES="jdk-8u101-linux-x64.tar.gz $FILES"
-
-if ! [[ -f jce_policy-8.zip ]]; then
-  while ! $(file jce_policy-8.zip | grep -q 'Zip archive data'); do
-    $CURL -O -L -C - -b "oraclelicense=accept-securebackup-cookie" http://download.oracle.com/otn-pub/java/jce/8/jce_policy-8.zip
-  done
-fi
-FILES="jce_policy-8.zip $FILES"
-
-# Pull all the (unversioned) gems required for the cluster 
-for i in patron wmi-lite simple-graphite ruby-augeas chef-rewind; do
-  if ! [[ -f gems/${i}.gem ]]; then
-    gem fetch ${i}
-    ln -s ${i}-*.gem ${i}.gem || true
-  fi
-  FILES="${i}*.gem $FILES"
-done
-
-# Get the Rubygem for mysql2
-if ! [[ -f gems/mysql2.gem ]]; then
-  gem fetch mysql2 -v 0.4.4
-  ln -s mysql2-*.gem mysql2.gem || true
-fi
-FILES="mysql2*.gem $FILES"
-
-# Get the Rubygem for kerberos
-if ! [[ -f gems/rake-compiler.gem ]]; then
-  gem fetch rake-compiler
-  ln -s rake-compiler*.gem rake-compiler.gem || true
-fi
-FILES="rake-compiler*.gem $FILES"
-
-# Get the Rubygem for sequel
-if ! [[ -f gems/sequel.gem ]]; then
-  gem fetch sequel -v 4.36.0
-  ln -s sequel-*.gem sequel.gem || true
-fi
-FILES="sequel*.gem $FILES"
-
-# Get the Rubygem for rkerberos
-if ! [[ -f gems/rkerberos.gem ]]; then
-  gem fetch rkerberos
-  ln -s rkerberos*.gem rkerberos.gem || true
-fi
-FILES="rkerberos*.gem $FILES"
-
-# Get the Rubygem for webhdfs
-if ! [[ -f gems/webhdfs.gem ]]; then
-  gem fetch webhdfs -v 0.5.5
-  ln -s webhdfs-*.gem webhdfs.gem || true
-fi
-FILES="webhdfs*.gem $FILES"
+if [ $(dpkg-query -W -f='${Status}' chefdk 2>/dev/null | grep -c 'ok installed') -eq 0 ]; then
```

Comment:
Are you looking for a count (`-c`) somehwere I am missing or just if the string is present (`-q`)?